### PR TITLE
Replace npm with yarn #11278

### LINKED ISSUE #11278: Use one package manager, not two
Now that NPM supports version locking in v5, it would be nice to only have a single package manager, e.g. replace yarn.lock with package-lock.json.

/to @danielrozenberg /cc @rsimha-amp @erwinmombay

### PR DESCRIPTION
**Pull request by danielrozenberg**
Fixes #11278

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
   "description": "Fastish HTML",
   "main": "index.js",
   "engines": {
-    "node": "^4.7.0",
-    "npm": "^5.0.0"
+    "node": ">=4.8.0",
```
Comment (rsimha-amp) [inline package.json]: I believe you should remove the `engines` section from this file. We specify the `node` version that Travis must use via `.travis.yml`, and Travis' built in version of `yarn` used to work fine as it was.

Comment (rsimha-amp) [reply package.json]: See https://github.com/travis-ci/travis-ci/issues/4653#issuecomment-169035161 for an explanation of why Travis ignores the versions of `node` and `npm` specified in `package.json`.

Comment (danielrozenberg) [reply package.json]: No change, Travis still fails on Yarn

**Comment by rsimha-amp** [comment]
@danielrozenberg, looks like the error in your latest commit is due to `yarn` complaining that it needs `node 6` in order to install `punycode`. It turns out that `punycode` was introduced as a dependency of `browserify`, which was installed as part of #9775.

And as luck would have it, @erwinmombay merged #9775 just after you had removed `yarn`, so this problem became yours instead of his :/

I encountered a similar version incompatibility in #10671, and at that time, I could get away with it. This time, I'd imagine you'd need to downgrade `babelify` to a version that uses a version of `punycode` that is compatible with `node 4`.

Dependencies are hard :(

**Comment by rsimha-amp** [comment]
Update: Let's move to node 6 on Travis. I'll follow up this PR with one that introduces a check to make sure `yarn` is used by all committers to install dependencies, and not `npm`.

**Comment by rsimha-amp** [comment]
@danielrozenberg FYI, I just tested your PR with node 6, and the yarn install was successful.

**Comment by danielrozenberg** [comment]
Aren't we forcing everyone to update to node 6 then? If yarn can't build amphtml with node 4.8, then changing the .travis.yml isn't going to be enough

**Comment by rsimha-amp** [comment]
Yes, everyone should upgrade to node 6. From yesterday's discussion, here's what needs to happen:

1. Update the version of node used on Travis from 4.8 to 6 (this PR)
2. Update the instructions to reflect that everyone must now use node 6 for local development (this PR)
3. Add a check to make sure that those who freshly clone the AMP repo use yarn to install their dependencies, and not npm (I'll send out a separate PR)
4. Send out a PSA to the team

**Comment by danielrozenberg** [comment]
Done 1 & 2, PTAL

File: build-system/SERVING.md [hunk 2]
```diff
@@ -18,17 +18,18 @@ limitations under the License.
 
 ## Requirements
 - git
-- node + npm
+- node 6+
 - gulp (installed globally)
 - java 8
+- yarn 1.0.2+ (see https://yarnpkg.com/)
```
Comment (rsimha-amp) [inline build-system/SERVING.md]: nit: Move this up the list, just below `node`, to reflect the order in which they are likely to be installed.

File: build-system/SERVING.md [hunk 3]
```diff
@@ -18,17 +18,18 @@ limitations under the License.
...
 
 ## Steps
 ```bash
 git clone https://github.com/ampproject/amphtml.git
 cd amphtml
 # Checkout a tag
 git checkout 123456789
-npm install
+yarn
```
Comment (rsimha-amp) [inline build-system/SERVING.md]: Does this need to be `yarn install`? Or is `yarn` sufficient if there's a `yarn.lock`?

File: contributing/getting-started-e2e.md [hunk 4]
```diff
@@ -151,20 +151,20 @@ Now run `git remote -v` again and notice that you have set up your upstream alia
...
-   npm install -g npm@latest
+   npm install -g yarn@latest
    ```
 
    The preceding command might require elevated privileges using `sudo` on some platforms.
 
 * In your local repository directory (e.g. `~/src/ampproject/amphtml`), install the packages that AMP uses by running
    ```
-   npm install
+   yarn
```
Comment (rsimha-amp) [inline contributing/getting-started-e2e.md]: Same question here, re: `yarn` vs. `yarn install`.

File: build-system/SERVING.md (hunk 3 above)
Comment (danielrozenberg) [reply build-system/SERVING.md]: "yarn" == "yarn install", regardless of yarn.lock or not

https://yarnpkg.com/lang/en/docs/cli/install/

File: contributing/getting-started-e2e.md (hunk 4 above)
Comment (danielrozenberg) [reply contributing/getting-started-e2e.md]: ACK

File: build-system/SERVING.md (hunk 2 above)
Comment (danielrozenberg) [reply build-system/SERVING.md]: Done

File: build-system/SERVING.md (hunk 3 above)
Comment (rsimha-amp) [reply build-system/SERVING.md]: Cool, thanks for checking!

**Comment by erwinmombay** [comment]
fyi @dknecht we bumped node requirement to >6 and that `yarn` should be used to install node deps

**Pull request merged**